### ISSUE TITLE
Adding new optimizer SinkSGD_adv, potentially powerful competition to Muon and AdamW

### ISSUE THREAD

**Issue by RicemanT** [67372a/LoRA_Easy_Training_Scripts#90]
Implementation is here https://github.com/Koratahiu/Advanced_Optimizers
Original papers: https://arxiv.org/pdf/2502.06742

This optimizer have been picking steam lately from Onetrainer/AI Toolkit users, and some of them was claiming to me that it's quite amazing for both LoRA and Finetuning. A bit curious about it, wouldn't mind a new sgd variant.

**Comment by khanghy1000** [comment]
If the `adv_optm` package needs to be updated, please consider tagging parameters with `_is_lora_A`, `_is_lora_B`, `_is_dora_scale`, `is_vector`, and `is_hidden` to make this UI compatible with adv_optm's new features like Spectral Scaling.

**Comment by Darudado** [comment]
I've been trying it as I included it on my local fork and it's pretty cool .

**Comment by 67372a** [comment]
@RicemanT I have added the additional optimizers to easy scripts. I haven't looked at the new features / parameters mentioned by @khanghy1000 yet.

**Comment by 67372a** [comment]
@khanghy1000 I have implemented tagging of parameters for lycoris network impls and sd_script network impls, give it a try and let me know how it goes.

**Comment by khanghy1000** [comment]
@67372a it works great. I trained a lora (lycoris network) for anima using `SinkSGD_adv` with `spectral_normalization=True`, `weight_decay=0.1`, `cautious_wd=True`, `lr=1e-3`, REX, bs=2 and the training was pretty stable.

**Comment by RicemanT** [comment]
@khanghy1000 I actually want to ask, does sinksgd really operate at 1e-3? one of the guys who recommend it to me over on AI Toolkit use that, while @Darudado told me he used e-5 lr range. This is pretty weird, why does the opt work at both low and really high range like this....

**Comment by khanghy1000** [comment]
@RicemanT I used 1e-3 because spectral norm requires a higher learning rate (and dim = alpha). Even 1e-3 feels a little low with it. Sinksgd might use a lr in the e-5 range without spectral norm. The author of adv_optm said on one trainer discord that sinksgd might use a lr 5 to 10 times larger than adamw.
Edit: 1e-3 without spectral norm (`spectral_normalization=False`) will definitely fry the lora.

**Comment by Darudado** [comment]
Considering the previous message, my local changes probably aren't using spectral. I'll stash my changes and try machina's impl with the above parameters. Thanks

**Comment by 67372a** [comment]
Thanks for the confirmation it is working. Closing.
